release-tools/prow.sh

```diff
@@ -652,103 +511,78 @@ find_deployment () {
-            kind load docker-image --name csi-prow $i:csiprow || die "could not load the $i:latest image into the kind cluster"
-        done
-    fi
-
-    if deploy_driver="$(find_deployment "$(pwd)/deploy")"; then
-        :
-    elif [ "${CSI_PROW_DRIVER_REPO}" = "none" ]; then
-        return 1
-    else
-        git_checkout "${CSI_PROW_DRIVER_REPO}" "${CSI_PROW_WORK}/csi-driver" "${CSI_PROW_DRIVER_VERSION}" --depth=1 || die "checking out CSI driver repo failed"
-        if deploy_driver="$(find_deployment "${CSI_PROW_WORK}/csi-driver/deploy")"; then
-            :
-        else
-            die "deploy.sh not found in ${CSI_PROW_DRIVER_REPO} ${CSI_PROW_DRIVER_VERSION}. To disable E2E testing, set CSI_PROW_DRIVER_REPO=none"
-        fi
-    fi
-
-    if [ "${CSI_PROW_DRIVER_CANARY}" != "stable" ]; then
-        images="$images IMAGE_TAG=${CSI_PROW_DRIVER_CANARY}"
-    fi
-    # Ignore: Double quote to prevent globbing and word splitting.
-    # It's intentional here for $images.
-    # shellcheck disable=SC2086
-    if ! run env "CSI_PROW_TEST_DRIVER=${CSI_PROW_WORK}/test-driver.yaml" $images "${deploy_driver}"; then
-        # Collect information about failed deployment before failing.
-        collect_cluster_info
-        (start_loggers >/dev/null; wait)
-        info "For container output see job artifacts."
-        die "deploying the CSI driver with ${deploy_driver} failed"
-    fi
+kubectl_apply () {
+  // TODO once this CRD is part of core replace it with  'kubectl apply -f $1 --validate=false'
+  curl $1 | sed '/annotations/ a  \   \ "api-approved.kubernetes.io": "https://github.com/kubernetes/kubernetes/pull/123456"' | kubectl apply -f - --validate=false
 }
 
-# Installs all nessesary snapshotter CRDs  
-install_snapshot_crds() {
-  # Wait until volumesnapshot CRDs are in place.
-  CRD_BASE_DIR="https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/${CSI_SNAPSHOTTER_VERSION}/config/crd"
-  kubectl apply -f "${CRD_BASE_DIR}/snapshot.storage.k8s.io_volumesnapshotclasses.yaml" --validate=false
-  kubectl apply -f "${CRD_BASE_DIR}/snapshot.storage.k8s.io_volumesnapshots.yaml" --validate=false
-  kubectl apply -f "${CRD_BASE_DIR}/snapshot.storage.k8s.io_volumesnapshotcontents.yaml" --validate=false
+# Installs all nessesary CRDs  
+install_crds() {
+  # Wait until cosi CRDs are in place.
+  CRD_BASE_DIR="https://raw.githubusercontent.com/kubernetes-sigs/container-object-storage-interface-api/master/crds"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketclasses.yaml"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketrequests.yaml"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_buckets.yaml"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketaccessclasses.yaml"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketaccessrequests.yaml"
+  kubectl_apply  "${CRD_BASE_DIR}/objectstorage.k8s.io_bucketaccesses.yaml"
```

Comment:
Since https://github.com/kubernetes-sigs/container-object-storage-interface-api/pull/13 there's a `kustomization.yaml` in https://github.com/kubernetes-sigs/container-object-storage-interface-api/tree/master/crds wich allows to deploy the CRDs (including required `labels`) without exhaustively listing all of them here.

You could use, as an example, `kustomize build "https://github.com/kubernetes-sigs/container-object-storage-interface-api/crds/?ref=master" | kubectl apply -f -`.